### PR TITLE
Added reset by phone number functionality

### PR DESCRIPTION
**Pull request by peterolayinka**
# Problem
This package is limited to recovering passwords by email only leaving out use case for where we have users signup with phone number.

# solution
Added functionality to recover passwords by phone number.

### CODE REVIEW

File: django_rest_passwordreset/serializers.py [hunk 1]
```diff
@@ -8,9 +8,28 @@
...
+    except phonenumbers.phonenumberutil.NumberParseException as e:
+        return False, e._msg
+
+class EmailPhoneSerializer(serializers.Serializer):
+    email = serializers.EmailField(required=False)
+    phone = serializers.CharField(required=False)
+
+    def validate(self, attrs):
+        if not attrs.get("email") and not attrs.get("phone"):
+            raise serializers.ValidationError('Please provide Email or Phone Number')
```
Comment (bruno-santiago) [inline django_rest_passwordreset/serializers.py]: This message doesn't make clear for user which fields needs to be provided.

```diff
-raise serializers.ValidationError('Please provide Email or Phone Number')
+raise serializers.ValidationError('Please provide email or phone field')
```

**Comment by Vayel** [comment]
Thanks for the PR! It would be convenient to pass the authentication method (email or phone) to the [signal's receiver](https://github.com/anexia-it/django-rest-passwordreset/blob/2b5afad1cc3096caf45b66cbee97c32b7c2f6af9/django_rest_passwordreset/views.py#L191).

**Comment by enricostano** [comment]
What would be really nice is the ability to define a custom serializer extending the library. For instance in our case we need to identify the user by VAT number.

**Comment by enricostano** [comment]
Exactly this approach is what I was hoping for https://github.com/anexia-it/django-rest-passwordreset/pull/93

@peterolayinka this would allow you to use phone number
@Vayel what do you think?

**Comment by Vayel** [comment]
@enricostano sounds good to me!

**Comment by peterolayinka** [comment]
Hi Guys,

Thanks for the review, I totally missed this conversation. I guess this is due to plenty of work-related emails from GitHub.

**Comment by enricostano** [comment]
@morantron do we still need this?

**Comment by Morantron** [comment]
> @Morantron do we still need this?

yep

**Comment by enricostano** [comment]
@anx-hnezbeda do you mind keeping this issue open? Thanks!

**Comment by peterolayinka** [comment]
> A universal lookup field as proposed in #93 would also cover phonenumbers, so I would recommend to merge #93 and reject this PR.

@Morantron @enricostano what approach would you prefer to have this merged?

Though, I made a fork already which I use in my codebase.

**Pull request closed without merging**